kubernetes/config-processor/README.md

```diff
@@ -1,3 +1,100 @@
 # Config Processor
 
-This is a simple image that is used to "process" many of the config maps produced by kubernetes. It simply mounts them as an input volume and writes them over to an output volume, but at the same time it scans all files for the text `[[URL:...]]`, e.g. `[[URL:https://mywebsite.com/mysequence.txt]]`. Where it finds that pattern it downloads the file from this URL and replaces the `[[URL:...]]` with the contents of the file. This allows us to deal with large reference genomes. 
+## Overview
+
+The Loculus Config Processor is a utility container designed to dynamically process configuration files in Kubernetes deployments. It serves as an init container that transforms configuration files by:
+
+1. Fetching and inserting external content from URLs
+2. Substituting environment variables for sensitive information (passwords, secrets)
+
+This allows configuration files to reference large external assets (like reference genomes) and inject sensitive data without storing them directly in ConfigMaps.
+
+## How It Works
+
+The Config Processor operates in three main steps:
+
+1. **Copy**: Duplicates the entire directory structure from input to output
+2. **Fetch URLs**: Finds and replaces `[[URL:https://example.com/file.txt]]` patterns with the actual content from those URLs
```

Comment:
Thanks! I would maybe add here that it can only do string replacement and can't add in yaml contents from a file

Reply:
👍 https://github.com/loculus-project/loculus/pull/4122/commits/b040240310785eeeba90aca83bfd0aa6f0f496e5